interfaces/repo_test.go

```diff
@@ -468,19 +436,25 @@ func (s *RepositorySuite) TestAddSlotFailsWhenInterfaceIsUnknown(c *C) {
 }
 
 func (s *RepositorySuite) TestAddSlotFailsWhenSlotNameIsInvalid(c *C) {
-	err := s.emptyRepo.AddSlot(&Slot{Snap: s.slot.Snap, Name: "bad-name-", Interface: s.slot.Interface})
+	slot := slotFromYaml(c, "bad-name-", []byte(`
+name: snap
+slots:
+    bad-name-: interface
+`))
+	err := s.emptyRepo.AddSlot(slot)
 	c.Assert(err, ErrorMatches, `invalid interface name: "bad-name-"`)
+	c.Assert(s.emptyRepo.AllSlots(""), HasLen, 0)
 }
 
 func (s *RepositorySuite) TestAddSlotFailsWithInvalidSnapName(c *C) {
-	slot := &Slot{
-		Snap:      "bad-snap-",
-		Name:      "name",
-		Interface: "interface",
-	}
-	err := s.testRepo.AddSlot(slot)
+	slot := slotFromYaml(c, "slot", []byte(`
```

Comment:
Also bogus now. It's the yaml parsing logic that should complain about the name.